### PR TITLE
executor: change the type of AggFunc.args and AggExec.groupByItem to []*Column

### PR DESCRIPTION
**Pull request by XuHuaiyu**
### What problem does this PR solve? 
Fix issue #7674 

Before this PR, the types of definitions for `arguments`
of AggFunc and AggExec.GroupByItem are defined as 
[]Expression. This PR change them to []*Column.

### What is changed and how it works?

After this PR, we can directly use `row.Getxxx` when 
evaluating AggFunc.Args and GroupByItems, thus avoid
the cost of function call when using Expression.Eval.

### Check List 

Tests 

exist tests

Code changes

 - Has exported function/method change
 - Has exported variable/fields change

Side effects
Related changes

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/7697)

### CODE REVIEW

**Comment by shenli** [comment]
Any benchmark result?

File: cmd/explaintest/r/tpch.result [hunk 1]
```diff
@@ -1287,15 +1287,15 @@ cntrycode
...
-    └─Projection_38	0.00	root	substring(tpch.customer.c_phone, 1, 2), tpch.customer.c_acctbal
-      └─Selection_39	0.00	root	not(26_aux_0)
-        └─HashLeftJoin_40	0.00	root	left outer semi join, inner:TableReader_46, equal:[eq(tpch.customer.c_custkey, tpch.orders.o_custkey)]
-          ├─Selection_41	0.00	root	in(substring(tpch.customer.c_phone, 1, 2), "20", "40", "22", "30", "39", "42", "21")
-          │ └─TableReader_44	0.00	root	data:Selection_43
-          │   └─Selection_43	0.00	cop	gt(tpch.customer.c_acctbal, NULL)
-          │     └─TableScan_42	7500000.00	cop	table:customer, range:[-inf,+inf], keep order:false
-          └─TableReader_46	75000000.00	root	data:TableScan_45
-            └─TableScan_45	75000000.00	cop	table:orders, range:[-inf,+inf], keep order:false
+Sort_33	1.00	root	custsale.cntrycode:asc
```
Comment (winoros) [inline cmd/explaintest/r/tpch.result]: Why plan id changed?

**Comment by shenli** [comment]
@XuHuaiyu Please address the comments and resolve the conflicts.

**Comment by shenli** [comment]
Any update?

**Pull request closed without merging**